### ISSUE TITLE
False-positive C0412 ungrouped-imports

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4360]
### Steps to reproduce

Given a file `a.py`:
```python
"""False-positive ungrouped-imports."""
from numpy import array
from pandas import DataFrame

if 2 > 1:
    from numpy import zeros

print(array, DataFrame, zeros)
```

### Current behavior

Result of `pylint a.py`:
```
D:\a.py:6:4: C0412: Imports from package numpy are not grouped (ungrouped-imports)

------------------------------------------------------------------
Your code has been rated at 8.00/10 (previous run: 6.00/10, +2.00)
```

### Expected behavior

Perfect score according to https://github.com/PyCQA/pylint/issues/4212#issuecomment-819539155.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.4
astroid 2.5.2
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AMD64)]
```

**Comment by hippo91** [comment]
@bersbersbers thanks for the report.

**Comment by bersbersbers** [comment]
I cannot reproduce this any longer.

```
pylint 2.12.2
astroid 2.9.3
Python 3.9.10 (main, Jan 17 2022, 14:25:44) 
[GCC 7.5.0]
```
